File: I. EASY/169. Majority Element/solution.cpp
class Solution
{
public:
    int majorityElement(vector<int> &nums)
    {
        const int n = nums.size();
        int count = 0, nb = 0;
        sort(nums.begin(), nums.end());

        for (int i = 0; i < n; i++)
        {
            if (nb == nums[i])
            {
                count++;
                if (count > n / 2)
                    return nums[i];
            }
            else
                nb = nums[i], count = 1;
        }

        return nums[0];
    }
};

File: I. EASY/169. Majority Element/solution.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        n = len(nums)
        hmap = {}

        for e in nums:
            # count the frequency of each element
            if e in hmap:
                hmap[e] += 1
            else:
                hmap[e] = 1
            # check if the element is majority without iterating through the whole list
            if hmap[e] > n / 2:
                return e

        # return the element with the highest frequency if the majority element is not found
        return max(hmap, key=hmap.get)
